pkg/cache/external/scheduler_cache.go

```diff
@@ -65,6 +65,16 @@ func (cache *SchedulerCache) GetNodesInfoMap() map[string]*schedulernode.NodeInf
 	return cache.nodesMap
 }
 
+func (cache *SchedulerCache) GetNodesInfoMapCopy() map[string]*schedulernode.NodeInfo {
+	cache.lock.RLock()
+	defer cache.lock.RUnlock()
+	copyOfMap := make(map[string]*schedulernode.NodeInfo, len(cache.nodesMap))
+	for k, v := range cache.nodesMap {
+		copyOfMap[k] = v.Clone()
```

Comment:
Do we need to clone the NodeInfo here? I think it is enough to just pass it: copyOfMap[k] = v